src/BenchmarkDotNet/Running/BenchmarkRunnerClean.cs

```diff
@@ -550,15 +596,21 @@ private static (BenchmarkRunInfo[], List<ValidationError>) GetSupportedBenchmark
         {
             List<ValidationError> validationErrors = new ();
 
-            var runInfos = benchmarkRunInfos.Select(info => new BenchmarkRunInfo(
-                    info.BenchmarksCases.Where(benchmark =>
-                    {
-                        var errors = benchmark.GetToolchain().Validate(benchmark, resolver).ToArray();
-                        validationErrors.AddRange(errors);
-                        return !errors.Any();
-                    }).ToArray(),
-                    info.Type,
-                    info.Config))
+            var runInfos = benchmarkRunInfos.Select(info =>
+                {
+                    var eventHandler = info.Config.GetCompositeEventHandler();
+                    return new BenchmarkRunInfo(
+                        info.BenchmarksCases.Where(benchmark =>
+                        {
+                            var errors = benchmark.GetToolchain().Validate(benchmark, resolver).ToArray();
+                            validationErrors.AddRange(errors);
+                            if (errors.Any())
+                                eventHandler.HandleUnsupportedBenchmark(benchmark);
```

Comment:
`OnValidaitonError` seems to be a better name (not every validation error is a hard stop)

Reply:
I have removed `OnUnsupportedBenchmark` since they surface in `OnValidationError` anyways